lib/nostrum/consumer.ex

```diff
@@ -406,7 +413,14 @@ defmodule Nostrum.Consumer do
         {:noreply, state}
       end
 
-      defoverridable child_spec: 1, start_link: 1, init: 1, handle_continue: 2, handle_info: 2
+      defoverridable GenServer
+    end
+  end
+
+  defmacro __before_compile__(_env) do
+    quote location: :keep do
+      @impl Nostrum.Consumer
+      def handle_event(_), do: :noop
```

Comment:
Is there any guarantee on where this will be inserted into the module?

If it would be inserted early then no events would be received in the consumer.

Reply:
> Is there any guarantee on where this will be inserted into the module?

Yes.

@before_compile is inserted at the end of the module.

Reply:
Thank you for the explanation 👍